csharp/src/SeedLang.Common/BlockRange.cs

```diff
@@ -0,0 +1,49 @@
+// Copyright 2021 The Aha001 Team.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace SeedLang.Common {
+  // Represents a range in a SeedBlock source code.
+  public class BlockRange : Range {
+    public static readonly BlockRange Empty = new BlockRange(null as BlockPosition);
```

Comment:
I understand the code here is using null as a null BlockPosition. If we didn't need distinguish Null BlockRange and null TextRange, we can have a NullRange class implement Range to indicate it cannot be associated with any code position. Then The BlockPosition and TextPosition can be struct without null. It can remove the null check for position objects.

Reply:
I double-checked on this and realized there is not urgent request on a NullRange or an EmptyRange for now. We can simply pass a null reference to the new Diagnostic instance as the Range argument. No need to maintain a special instance or a special subclass.